Support use of an external copy of the PCRE library:

* configure.in: Set abs_{builddir,srcdir} higher.  Add --with-pcre
flag; build against external PCRE library if used.

* Makefile.in (install-include): Don't install pcre headers any more.

* srclib/Makefile.in (SUBDIRS): Remove.

PR: 27550 (part two)
Submitted by: Andres Salomon <dilinger voxel.net>, Joe Orton

Index: httpd-2.0.59/configure.in
===================================================================
--- httpd-2.0.59.orig/configure.in
+++ httpd-2.0.59/configure.in
@@ -124,16 +124,47 @@ dnl PCRE and for our config tests will b
 AC_PROG_CC
 AC_PROG_CPP
 
-echo $ac_n "${nl}Configuring PCRE regular expression library ...${nl}"
+dnl Absolute source/build directory
+abs_srcdir=`(cd $srcdir && pwd)`
+abs_builddir=`pwd`
+
+AC_ARG_WITH(pcre,
+APACHE_HELP_STRING(--with-pcre=PATH,Use external PCRE library))
+
+case $with_pcre in
+yes) AC_PATH_PROG(PCRE_CONFIG, pcre-config, false) ;;
+ /*) if test -d "$with_pcre" && test -x "$with_pcre/bin/pcre-config"; then
+       PCRE_CONFIG=$with_pcre/bin/pcre-config
+     elif test -x "$with_pcre"; then
+       PCRE_CONFIG=$with_pcre
+     fi
+
+     if $PCRE_CONFIG --version >&/dev/null; then :; else
+       AC_MSG_ERROR([Did not find pcre-config script at $PCRE_CONFIG])
+     fi
+     ;;
+*) PCRE_CONFIG=false ;;
+esac
+
+if test "$PCRE_CONFIG" != "false"; then
+  AC_MSG_NOTICE([Using external PCRE library from $PCRE_CONFIG])
+  APR_ADDTO(CFLAGS, [`$PCRE_CONFIG --cflags`])
+  APR_ADDTO(LIBS, [`$PCRE_CONFIG --libs`])
+else
+  # Build the bundled PCRE
+  AC_MSG_NOTICE([Configuring PCRE regular expression library])
 
-APR_SUBDIR_CONFIG(srclib/pcre,
+  APR_SUBDIR_CONFIG(srclib/pcre,
                   [--prefix=$prefix --exec-prefix=$exec_prefix --libdir=$libdir --includedir=$includedir --bindir=$bindir])
 
-echo $ac_n "${nl}Configuring Apache httpd ...${nl}"
+  APR_ADDTO(AP_LIBS, [$abs_builddir/srclib/pcre/libpcre.la])
+  APR_ADDTO(CPPFLAGS, [-I$abs_builddir/srclib/pcre])
 
-dnl Absolute source/build directory
-abs_srcdir=`(cd $srcdir && pwd)`
-abs_builddir=`pwd`
+  AP_BUILD_SRCLIB_DIRS="$AP_BUILD_SRCLIB_DIRS pcre"
+  AP_CLEAN_SRCLIB_DIRS="$AP_CLEAN_SRCLIB_DIRS pcre"
+fi
+
+echo $ac_n "${nl}Configuring Apache httpd ...${nl}"
 
 dnl If the source dir is not equal to the build dir, 
 dnl then we are running in VPATH mode.
@@ -484,9 +515,6 @@ AC_ARG_WITH(suexec-umask,
 APACHE_HELP_STRING(--with-suexec-umask,umask for suexec'd process),[
   AC_DEFINE_UNQUOTED(AP_SUEXEC_UMASK, 0$withval, [umask for suexec'd process] ) ] )
 
-dnl AP_LIBS specifies the actual libraries. note we have some required libs.
-AP_LIBS="$abs_builddir/srclib/pcre/libpcre.la $AP_LIBS"
-
 dnl APR should go after the other libs, so the right symbols can be picked up
 AP_LIBS="$AP_LIBS `$apu_config --link-libtool --libs` `$apr_config --link-libtool --libs`"
 APACHE_SUBST(AP_LIBS)
